### ISSUE TITLE
Optional parameters not optional in this module

### ISSUE THREAD

**Issue by Remco75** [ionic-team/ionic-native#1939]
**I'm submitting a ...**  (check one with "x")
[ x] bug report
[ ] feature request

**Current behavior:**
When using the cordova-music-controls-plugin with ionic native all options are required in the type.
in this example it clearly states that ios options are optional:
` MusicControls.create({
    track       : 'Time is Running Out',		// optional, default : ''
	artist      : 'Muse',						// optional, default : ''
    cover       : 'albums/absolution.jpg',		// optional, default : nothing
	// cover can be a local path (use fullpath 'file:///storage/emulated/...', or only 'my_image.jpg' if my_image.jpg is in the www folder of your app)
	//			 or a remote url ('http://...', 'https://...', 'ftp://...')
	isPlaying   : true,							// optional, default : true
	dismissable : true,							// optional, default : false

	// hide previous/next/close buttons:
	hasPrev   : false,		// show previous button, optional, default: true
	hasNext   : false,		// show next button, optional, default: true
	hasClose  : true,		// show close button, optional, default: false

	// iOS only, optional
	album       : 'Absolution',     // optional, default: ''
	duration : 60, // optional, default: 0
	elapsed : 10, // optional, default: 0
  	hasSkipForward : true, //optional, default: false. true value overrides hasNext.
  	hasSkipBackward : true, //optional, default: false. true value overrides hasPrev.
  	skipForwardInterval : 15, //optional. default: 0.
	skipBackwardInterval : 15, //optional. default: 0. 

	// Android only, optional
	// text displayed in the status bar when the notification (and the ticker) are updated
	ticker	  : 'Now playing "Time is Running Out"'
}, onSuccess, onError);
`
But leaving for instance "duration" out will throw an typescript type error. And I do not need those fields

**Expected behavior:**
The optional fields should be , well, optional ;-)

**Steps to reproduce:**

**Related code:**

```
insert any relevant code here
```

**Other information:**

**package.json info:**

```json
insert the content here
```

**Comment by Remco75** [comment]
Is any one out there? it seems that the native repository is near dead when it comes to reacting to issues.

**Comment by fdambrosio** [comment]
Docs are wrong, you're right...anyone out there?

**Comment by Remco75** [comment]
well, there's someone out there, but they do not react to issues until they decide to dom something I guess.. (see #2159  )

**Comment by fdambrosio** [comment]
about Ionic Native Music Controls there's new parameters added on cordova-music-controls-plugin  2.1.4 


```
hasScrubbing : false, //optional. default to false. Enable scrubbing from control center progress bar 
playIcon: 'media_play',
pauseIcon: 'media_pause',
prevIcon: 'media_prev',
nextIcon: 'media_next',
closeIcon: 'media_close',
notificationIcon: 'notification'
```

**Comment by joshbabb** [comment]
So, one of the issues is that a fix needs to be proposed and a PR opened for anything to happen.  Music controls are probably low priority... but let me take a crack at this real quick.

**Comment by joshbabb** [comment]
OK, @Remco75, I hope that PR #2186 solves your issue!  Please pull, build, and test it to confirm or deny that.

The magic sauce is that the `?` in `variable?: <type>` denotes that a variable is optional.

**Comment by joshbabb** [comment]
Erp, @ghenry22 already submitted #2094 ... test that please, eh?

**Comment by Remco75** [comment]
Tested: this is fixed with #2094 .thnx

**Comment by Remco75** [comment]
I will close this one now
